File: 0217-contains-duplicate/0217-contains-duplicate.cs
public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        if(nums.Length == 0 || nums == null) return false;
        
        HashSet<int> hs = new HashSet<int>();
        
        foreach(int n in nums){
            if(!hs.Add(n)) return true;
        }
        
        return false;
    }
}

File: 0217-contains-duplicate/0217-contains-duplicate.ts
function containsDuplicate(nums: number[]): boolean {
    const numsMap = new Map();
    for(let i=0; i < nums.length; i++){
        if(numsMap.has(nums[i])) return true;
        numsMap.set(nums[i], i);
    }
    return false;
};